src/Proto.Cluster/Partition/MemberHashRing.cs

```diff
@@ -0,0 +1,22 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="MemberRing.cs" company="Asynkron AB">
+//      Copyright (C) 2015-2021 Asynkron AB All rights reserved
+// </copyright>
+// -----------------------------------------------------------------------
+using System.Collections.Generic;
+using Proto.Router;
+
+namespace Proto.Cluster.Partition
+{
+    public class MemberHashRing : HashRing<Member>
+
+    {
+        public MemberHashRing(IReadOnlyCollection<Member> nodes) : base(nodes, member => member.Address, MurmurHash2.Hash, 50)
```

Comment:
My understanding of the stickiness is that the more replicas, the bigger the risk of not being sticky. e.g. more parts of the hash ring are moving.
So some config might be nice

Reply:
The one potential problem with making it configurable is that two different configs can have different answers for which member an identity belongs to. If both configurations exists in a cluster, the behavior would be incorrect, and you could end up with lots of weird failure cases.  It might be slightly less error prone to choose a constant default value. 
The downside would be for someone who would like to tune the  distribution / stickyness. An option could be to make the partition identity check if the other nodes have the same settings as itself, and otherwise fail fast on startup.

Reply:
> make the partition identity check if the other nodes have the same settings

The same problem would occur if we were to change the constant replica count between versions, so maybe this is something we'd want to handle either way?

This got me thinking, is there any way to handle a change to the replica count gracefully? If all nodes must be in agreement on the replica count, then as long as there's one node alive then the value can't change. Meaning the only way to change it would be to shut down the whole cluster first, right?

Reply:
That is actually a very good point. The first node active in the cluster could be the master for replica count, and additional nodes would use that instead of any local settings. Then it can be tuned without worrying about it being out of sync across the cluster.

Reply:
A question then is what happens when the leader decides to change the replica count. Would the cluster become in sync quickly enough to avoid disruptions? Or would it be necessary to have some sort of coordinated pause until all nodes use the same value?

Reply:
I definitely don't think that needs to be dynamic at all. Keep the initial value from starting the cluster, no problem with updates.

Reply:
Is this a likely scenario though?
Under what conditions would a node decide that hey, let's tweak the replicas of the hash ring?

Reply:
Let me rephrase, because I botched my previous comment a bit 😅 

If 1) the replicaCount value changes (either because the const value changed or because it was changed via config) and 2) nodes always check what the current value in the cluster is (controlled by the oldest node), then wouldn't it mean the only way to change the value is to shut down all nodes? Even if you bring it down to a single-node cluster, that node would already be configured to use the old value. Or am I missing something?

Reply:
No, I think you summarized it :) The cluster will have to be shut down if you want to change replica count, if we go for this solution where we allow the first node to decide the setting.
It then later being removed from the cluster cannot change it, as that would break the behavior between the rest of the nodes.